lib/main/server.js

```diff
@@ -19,39 +20,42 @@ Server.prototype.processRequest = function(request, response) {
   var urlParams = url.parse(request.url, true);
   if (request.method === 'GET' && urlParams.pathname === '/I/want/title/') {
     var addresses = (typeof(urlParams.query.address) === 'string') ? [urlParams.query.address] : urlParams.query.address;
-    var titles = [];
     var body = [];
 
-    function processTitle(title, index){
-      titles.push(title);
-      if(index < addresses.length - 1){
-          console.log(this);
-          self.fetchTitle(addresses[index + 1], index + 1, processTitle);
-      } else {
-        response.statusCode = 200;
-        response.write("<html>");
-        response.write("<head></head>");
-        response.write("<body>");
-        response.write("<ul>");
-        response.write(titles.map(function(title){ return "<li>" + title + "</li>"; }).join(' '));
-        response.write("</ul>");
-        response.write("</body>");
-        response.write("</html>");
-        response.end();
-      }
-    }
-    self.fetchTitle(addresses[0], 0, processTitle);
+    var promises = addresses.map(function(address){
+      return self.fetchTitle(address);
+    });
+
+    RSVP.all(promises).then(function(titles){
+      response.statusCode = 200;
+      response.write("<html>");
+      response.write("<head></head>");
+      response.write("<body>");
+      response.write("<ul>");
+      response.write(titles.map(function(title){ return "<li>" + title + "</li>"; }).join(' '));
+      response.write("</ul>");
+      response.write("</body>");
+      response.write("</html>");
+      response.end();
+    }).catch(function(error){
+      response.statusCode = 500;
+      response.end();
+      console.log("Something went wrong..." + error)
+    });
+
   } else {
     response.statusCode = 404;
     response.end();
   }
 }
 
-Server.prototype.fetchTitle = function(address, index, callback) {
-  new Scrapper(address, function(result){
-    callback(result.url + " - " + result.dom('title').text(), index);
-  }, function(error){
-    callback(error.url + " - NO RESPONSE", index);
+Server.prototype.fetchTitle = function(address) {
+  return new RSVP.Promise(function(fullfill, reject){
+    Scrapper(address).then(function(result){
+      fullfill(result.url + " - " + result.dom('title').text());
+    }, function(error){
+      reject(error.url + " - NO RESPONSE");
```

Comment:
This will also introduce a bug as your Promises.all will resolve once all are fulfilled or 'got rejected' when at least one promise is rejected in the array.